### PR TITLE
Linux101: editor

### PR DESCRIPTION
**Pull request by core-man**
Related to #67.

**vim**
- Usage
  - https://github.com/vim/vim
  - https://learnxinyminutes.com/docs/zh-cn/vim-cn/
  - https://www.runoob.com/linux/linux-vim.html
  - https://missing-semester-cn.github.io/2020/editors/
  - http://billie66.github.io/TLCL/book/chap13.html
  - http://geophysics.eas.gatech.edu/classes/Intro_Unix/
  - https://zhuanlan.zhihu.com/p/68111471
  - https://coolshell.cn/articles/5426.html
- Setting
  - http://www.ruanyifeng.com/blog/2018/09/vimrc.html
  - https://sourabhbajaj.com/mac-setup/Vim/
  - https://www.jianshu.com/p/a0b452f8f720
  - https://vimawesome.com/
  - https://www.cnblogs.com/devilmaycry812839668/p/13170726.html
  - https://www.w3cschool.cn/vim/278q1hsu.html
  - https://www.w3cschool.cn/vim/47zk1ht5.html

**VS Code**
- Basics
  - https://code.visualstudio.com/
  - https://code.visualstudio.com/docs
- Remote-SSH
  - https://code.visualstudio.com/docs/remote/ssh
  - https://code.visualstudio.com/docs/remote/ssh-tutorial
  - https://code.visualstudio.com/docs/remote/troubleshooting#_quick-start-using-ssh-keys
  - https://kebingzao.com/2020/02/26/vscode-remote-ssh/
  - https://man.linuxde.net/ssh-keygen
  - https://man.linuxde.net/ssh-copy-id

### CODE REVIEW

**Comment by core-man** [comment]
I haven't found a good vim manual yet. Meanwhile, I am not sure if we should have a detailed vim manual here.

I am also finding some good vim settings and plugins so that we can make this section similar to OhMyZsh.

**Comment by seisman** [comment]
> Meanwhile, I am not sure if we should have a detailed vim manual here.

Definitely no. I would encourage users to use VSCode instead. VSCode now already can edit files on servers, so vim is less necessary now.

**Comment by core-man** [comment]
>  VSCode now already can edit files on servers.

Really, good to know that. So we can only leave the first few paragraphs.

**Comment by seisman** [comment]
> > VSCode now already can edit files on servers.
> 
> Really, good to know that. So we can only leave the first few paragraphs.

https://code.visualstudio.com/docs/remote/ssh-tutorial
https://kebingzao.com/2020/02/26/vscode-remote-ssh/

The tutorials look long, but I think they can be much shorter. You may try it if you have a Linux server to try.

**Comment by core-man** [comment]
> > > VSCode now already can edit files on servers.
> > 
> > 
> > Really, good to know that. So we can only leave the first few paragraphs.
> 
> https://code.visualstudio.com/docs/remote/ssh-tutorial
> https://kebingzao.com/2020/02/26/vscode-remote-ssh/
> 
> The tutorials look long, but I think they can be much shorter. You may try it if you have a Linux server to try.

Okay. We have a server.

**Comment by core-man** [comment]
> > > VSCode now already can edit files on servers.
> > 
> > 
> > Really, good to know that. So we can only leave the first few paragraphs.
> 
> https://code.visualstudio.com/docs/remote/ssh-tutorial
> https://kebingzao.com/2020/02/26/vscode-remote-ssh/
> 
> The tutorials look long, but I think they can be much shorter. You may try it if you have a Linux server to try.

I can use VS CODE to edit codes on the server. Look Great~

File: source/computer/editor.rst [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+           User seismo-learn
+           HostName 192.168.1.100
+           IdentityFile ~/.ssh/id_rsa-remote-ssh
+
+   其中，第一列是关键词，如 Host、User、HostName、IdentityFile，第二列是对应的值。seismology 是
+   用户自定义的远程主机标识，其他三个值同上文。可以按照该格式，向该文件中添加多个远程计算机。
+
+   然后点击左下角的远程连接图标，选择“Remote-SSH: Connect to Host”，点击“seismology”就可以登录远程计算机了。
+   也可以通过“远程资源管理器”，点击“seismology”登录远程计算机。
+
```
Comment (core-man) [inline source/computer/editor.rst:127]:
```suggestion

.. tip::

   如果远程计算机使用的 Shell 是 Bash，本地计算机是 Zsh，则可能无法启动 VS Code 的终端，
   需要修改一下配置文件。打开命令面板，输入 Remote-SSH: Settings 后，搜索 terminal.integrated.shell.linux。
   然后将 "/bin/zsh" 改为 "/bin/bash" 即可。详情请参考 `microsoft/vscode-remote-release issues #38 <https://github.com/microsoft/vscode-remote-release/issues/38>`__

```

File: source/computer/editor.rst [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+`ssh <https://code.visualstudio.com/docs/remote/ssh>`__
+了解详细用法，本文简要介绍如何安装和使用该扩展包。
+
+在“扩展”中，搜索“Remote - SSH”，点击安装。安装完之后，左下角导航栏会多一个类似 ``><`` 的远程连接图标。
+之后可以通过点击该图标来使用该扩展包。
+
+接下来需要设置远程计算机的 SSH 主机。远程计算机一般已安装 SSH 服务器，本地计算机一般也已安装 SSH 客户端。
+我们还需要配置基于密钥的认证，这也是 VS Code 官方推荐的认证方式。本地计算机下，运行以下命令::
+
+    $ ssh-keygen -t rsa -b 4096 -f ~/.ssh/id_rsa-remote-ssh
```
Comment (core-man) [inline source/computer/editor.rst:90]:
```suggestion
    $ ssh-keygen -t rsa -f ~/.ssh/id_rsa-remote-ssh
```

File: source/computer/editor.rst [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+操作模式
+^^^^^^^^^
+
+Vim 的设计以大多数时间都花在阅读、浏览和进行少量编辑改动为基础，因此它具有多种操作模式：
+
+- 正常模式：移动光标、复制、粘贴、删除等
+- 插入模式（INSERT）：插入文本
+- 命令模式：执行一些指令
+- 替换模式（REPLACE）：替换文本
+- 可视化模式：用于选中文本块，分为可视化（VISUAL）、可视化行（VISUAL LINE）、可视化块（VISUAL BLOCK）
```
Comment (seisman) [inline source/computer/editor.rst:172]: Remove these two mode?

File: source/computer/editor.rst [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+在不同的操作模式下，键盘敲击的含义也不同。vim 默认会在左下角显示当前的操作模式。启动时的默认模式是正常模式，
+可以使用以下方式，在不同模式只见进行切换：
+
+- 按下 :kbd:`Esc` 键可以从任何其他模式返回正常模式
+- 在正常模式下，按 :kbd:`i` 键进入插入模式
+- 在正常模式下，按 :kbd:`:` 键进入命令模式
+- 在正常模式下，按 :kbd:`Shift` + :kbd:`R` 键进入替换模式
+- 在正常模式下，按 :kbd:`v` 键进入可视化（一般）模式
+- 在正常模式下，按 :kbd:`Shift` + :kbd:`v` 键进入可视化（行）模式
+- 在正常模式下，按 :kbd:`Ctrl` + :kbd:`v` 键进入可视化（块）模式
```
Comment (seisman) [inline source/computer/editor.rst:183]: Remove?

File: source/computer/editor.rst [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+在正常模式下，可以使用以下常用移动命令在文本中快速移动：
+
+- 基本移动：\ :kbd:`h`\（左）、\ :kbd:`j`\（下）、\ :kbd:`k`\（上）、\ :kbd:`l`\（右）。也可以使用上下左右键
+- 词：\ :kbd:`w`\ （下一个词）、\ :kbd:`b`\ （词初）、\ :kbd:`e`\ （词尾）
+- 行： \ :kbd:`0`\ （行初）、\ :kbd:`^`\ （第一个非空格字符）、\ :kbd:`$`\ （行尾）
+- 屏幕：\ :kbd:`Shift` + :kbd:`h`\ （屏幕首行）、\ :kbd:`Shift` + :kbd:`m`\ （屏幕中间）、\ :kbd:`Shift` + :kbd:`l`\ （屏幕底部）
+- 翻页：\ :kbd:`Ctrl` + :kbd:`u`\ （上翻）、\ :kbd:`Ctrl` + :kbd:`d`\ （下翻）
+- 文件：\ :kbd:`gg`\ （文件头）、\ :kbd:`Shift` + :kbd:`g`\ （文件尾）
+- 行数：{number} + :kbd:`Shift` + :kbd:`g` 会移动至行号为 number（如 20）的行（第 20 行）
```
Comment (seisman) [inline source/computer/editor.rst:214]: I've never used these keys.

Comment (core-man) [reply source/computer/editor.rst]: Remove some in 870c7ad. I keep three useful move commands.

File: source/computer/editor.rst (hunk 4 above)
Comment (core-man) [reply source/computer/editor.rst]: Remove some in 870c7ad.

File: source/computer/editor.rst (hunk 3 above)
Comment (core-man) [reply source/computer/editor.rst]: Remove some in 870c7ad.

File: source/computer/editor.rst [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+   :container: + shadow
+   :title: bg-info text-white font-weight-bold
+
+   VS Code 的界面默认显示语言是英文，可以安装中文语言包。点击菜单栏“查看”后选择“命令面板”
+   （快捷键：\ :kbd:`Ctrl` + :kbd:`Shift` + :kbd:`p`\ ），接着输入“configure display language”并按
+   :kbd:`Enter` 键，然后选择“安装其他语言”。这时界面会跳转到插件商店自动搜索其他语言，一般第一个就是中文，
+   即“Chinese (Simplified) Language Pack for Visual Studio Code”，点击安装就行了。
+   安装完之后自动重启，界面就变成中文了。
+
+编辑远程计算机文件
```
Comment (seisman) [inline source/computer/editor.rst:85]: Perhaps we should hide the whole section? Most people don't use servers?

Comment (core-man) [reply source/computer/editor.rst]: Yes~ Done in 8ce65d4.

**Comment by core-man** [comment]
I will further polish the first two subsections.

**Comment by seisman** [comment]
Perhaps this should be a section, not a subsection. Put it after "常用命令"?

image

**Comment by core-man** [comment]
> Perhaps this should be a section, not a subsection. Put it after "常用命令"?
> 
> image

Yes to me.

**Comment by core-man** [comment]
> Perhaps this should be a section, not a subsection. Put it after "常用命令"?

Done in 709121a.

**Comment by core-man** [comment]
I have further polished the first two paragraphs in a904809.

File: source/computer/editor.rst [hunk 7]
```diff
@@ -0,0 +1,244 @@
+编辑器
```
Comment (seisman) [inline source/computer/editor.rst:1]: 文本编辑器 or 代码编辑器?

File: source/computer/editor.rst [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+
+:本节贡献者: |姚家园|\（作者）、
+             |田冬冬|\（审稿）
+:最近更新日期: 2021-02-13
+:预计花费时间: 30 分钟
+
+----
+
+在终端中输入命令即可交给系统执行，但要编辑普通文件、脚本或者程序，则需使用一个叫做\ **编辑器**\ 的程序。
+编辑器是一个允许在屏幕上移动光标并编辑文字的程序，它是科研人员编写代码和和管理系统配置文件的重要工具。
```
Comment (seisman) [inline source/computer/editor.rst:12]:
```suggestion
不管是编辑普通文件、脚本还是程序源码，都需使用一个叫做\ **编辑器**\ 的程序。
Windows 自带的编辑器是记事本（Notepad），macOS 自带的编辑器是 TextEdit，
Linux 自带的编辑器一般是 Gedit。日常科研中，不建议使用这些编辑器，而建议
学习并掌握其他功能更强大更高效的编辑器。
```

File: source/computer/editor.rst [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+现代编辑器都是一个复杂且强大的工具，永远有新东西可学：\ **学的越多，效率越高**\ 。
+
+选择编辑器
+----------
+
+当前存在许多优秀的编辑器，该使用哪个呢？这个问题并没有标准答案。选择其中一个一直用下去，遇到需要重复劳动或者不顺心的地方时，
+就去寻找解决办法，如更新配置和安装插件。有了足够多的知识和经验后，可以选择是否继续使用老编辑器，或尝试和更换新编辑器。
+
+按照操作方式，编辑器可分为两种类型：基于命令行的和图形化的。基于命令行的编辑器一般启动快，操作高效，学习成本较低；
+图形化编辑器启动慢一点，学习成本较高，但功能更加强大。
```
Comment (seisman) [inline source/computer/editor.rst:40]: > 基于命令行的编辑器一般启动快，操作高效，学习成本较低；图形化编辑器启动慢一点，学习成本较高，但功能更加强大。

I can't agree with you. Most CLI editors are difficult to learn.

Comment (core-man) [reply source/computer/editor.rst]: Done in eaaf45a.

> 按照操作方式，编辑器可分为两种类型：基于命令行的和图形化的。基于命令行的编辑器一般启动较快，操作高效；图形化编辑器启动慢一点，但功能更加强大。

**Comment by core-man** [comment]
@seisman  we may also recommend Jupyter-Lab?

File: source/computer/editor.rst [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+  以便在无法使用 VS Code 或编辑简单的文本/代码时使用
+- Linux 系统一般自带了一个叫做 gedit 的图形化编辑器，但是使用该编辑器来写程序效率太低，\ **不建议**\ 使用
+
+VS Code
+-------
+
+可以参考\ :doc:`setup`\ 中相应操作系统下的“日常软件”安装 VS Code。
+
+刚接触 VS Code 的用户可以参考 VS Code 官方入门教程 `Get Started <https://code.visualstudio.com/docs/getstarted/introvideos>`__
+学习常用用法。日常科研中，随时根据个人习惯配置 VS Code、安装和使用扩展包等。
```
Comment (seisman) [inline source/computer/editor.rst:71]: This site provides YouTube videos, and not accessible to users in China.

File: source/computer/editor.rst [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+
+本文只介绍了 vim 的一些常见用法。
+
+操作模式
+^^^^^^^^^
+
+Vim 的设计以大多数时间都花在阅读、浏览和进行少量编辑改动为基础。因此，它具有多种操作模式，以下只介绍几种常用模式：
+
+- 正常模式：移动光标、复制、粘贴、删除等
+- 插入模式（INSERT）：插入文本
```
Comment (seisman) [inline source/computer/editor.rst:193]:
```suggestion
- 插入模式：插入文本
```

**Review by seisman — APPROVED** [review]

File: source/computer/editor.rst [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+需要注意的是，编写代码时，经常在文件间进行切换，并阅读、浏览和修改代码，而不是连续编写一大段的文字，因此编写代码和
+编辑一长段纯文本是很不一样的。编写程序/软件时，还需要进行编译、链接、调试、版本控制等。因此，纯粹的代码编辑器和 IDE 是有所区别的，
+前者主要是让我们优雅地编写代码，后者集成了诸如编译、链接、调试、版本控制等功能，使得开发程序（往往是大型软件）更加高效。
+
+对于刚接触编辑器的读者，我们根据个人经验推荐一些编辑器。熟练掌握这些编辑器后，可以尝试和使用其他更适合的编辑器。
+
+- VS Code 是目前最流行的代码编辑器或 IDE，启动较快，编辑文本也非常高效。\ **推荐**\ 在日常科研中使用 VS Code 编辑文本和代码
+- vim 是目前最流行的基于命令行的文本或代码编辑器，有时候可能是用户唯一可以使用的编辑器。\ **推荐**\ 掌握 vim 的基本编辑功能，
+  以便在无法使用 VS Code 或编辑简单的文本/代码时使用
+- Linux 系统一般自带了一个叫做 gedit 的图形化编辑器，但是使用该编辑器来写程序效率太低，\ **不建议**\ 使用
```
[the remark this answers is not in the document]
Comment (core-man) [reply source/computer/editor.rst:63]: We already make this comment in the first paragraph. So removed.

Comment (core-man) [inline source/computer/editor.rst:63]:
```suggestion
```

[the remark this answers is not in the document]
Comment (seisman) [reply source/computer/editor.rst]: OK

**Comment by core-man** [comment]
> @seisman we may also recommend Jupyter-Lab?

@seisman ??

**Comment by seisman** [comment]
No, jupyterlab is only for python. We should only mention it in the Python section.

File: source/computer/editor.rst (hunk 10 above)
Comment (core-man) [reply source/computer/editor.rst:70]: You mean the videos cannot be accessible to users in China or the website?

**Comment by core-man** [comment]
> No, jupyterlab is only for python. We should only mention it in the Python section.

Okay~

File: source/computer/editor.rst (hunk 10 above)
Comment (seisman) [reply source/computer/editor.rst:70]: The videos.

Comment (core-man) [reply source/computer/editor.rst:70]: If only the videos, new babies can still read the website.

Comment (seisman) [reply source/computer/editor.rst:70]: I don't expect any Chinese users will read the lengthy VSCode documentation.

Comment (core-man) [reply source/computer/editor.rst:70]: I cannot find any good documentation for VS Code Get Started except the following one in VS Code help.

`- 点击 VS Code 菜单栏“帮助”，选择相关选项。例如，“交互式演练场”、“文档”（Visual Studio Code 官方文档）、“键盘快捷键参考”等`

Comment (core-man) [reply source/computer/editor.rst:70]: We may remove the above VS Code Get Started?

Comment (seisman) [reply source/computer/editor.rst:70]: Perhaps people can just search for some tutorials.

Comment (seisman) [reply source/computer/editor.rst:70]: > We may remove the above VS Code Get Started?

Yes to me.

Comment (core-man) [reply source/computer/editor.rst]: Removed in fc55144.

**Comment by core-man** [comment]
I will merge it tonight after I finally check the typing.

**Review by seisman — APPROVED** [review]

File: source/computer/editor.rst [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+             |田冬冬|\（审稿）
+:最近更新日期: 2021-02-13
+:预计花费时间: 30 分钟
+
+----
+
+不管是编辑文本文件、脚本还是程序源码，都需使用一个叫做\ **编辑器**\ 的程序。
+Windows 自带的编辑器是记事本（Notepad），macOS 自带的编辑器是 TextEdit，
+Linux 自带的编辑器一般是 Gedit。日常科研中，不建议使用这些编辑器，而建议
+学习并掌握其他功能更强大更高效的编辑器。
```
Comment (core-man) [inline source/computer/editor.rst:14]:
```suggestion
学习并掌握其他具有更强大、更高效功能的编辑器。
```

File: source/computer/editor.rst [hunk 14]
```diff
@@ -0,0 +1,244 @@
+编辑器
+=======
+
+:本节贡献者: |姚家园|\（作者）、
+             |田冬冬|\（审稿）
+:最近更新日期: 2021-02-13
```
Comment (core-man) [inline source/computer/editor.rst:6]:
```suggestion
:最近更新日期: 2021-03-01
```

File: source/computer/editor.rst [hunk 15]
```diff
@@ -0,0 +1,244 @@
...
+
+当前存在许多优秀的编辑器，该使用哪个呢？这个问题并没有标准答案。选择其中一个一直用下去，遇到需要重复劳动或者不顺心的地方时，
+就去寻找解决办法，如更新配置和安装插件。有了足够多的知识和经验后，可以选择是否继续使用老编辑器，或尝试和更换新编辑器。
+
+按照操作方式，编辑器可分为两种类型：基于命令行的和图形化的。基于命令行的编辑器一般启动较快，操作高效；图形化编辑器启动慢一点，
+但功能更加强大。
+
+按照功能，编辑器又大致可分为三种类型：文本编辑器、代码编辑器以及集成开发环境（Integrated Development Environment，简称 IDE）。
+顾名思义，文本编辑器主要用于编写一大段文本，代码编辑器主要用于编写代码，IDE 主要用于开发大型软件。
+当然，一般的功能性分类主要考虑该编辑器更适用（即更高效、更快捷）做哪种编辑工作，不代表其不能用于其他工作。
```
Comment (core-man) [inline source/computer/editor.rst:46]:
```suggestion
按照功能进行分类主要考虑该编辑器更适用（即更高效、更快捷）做哪种编辑工作，不代表其不能用于其他工作。
```

File: source/computer/editor.rst [hunk 16]
```diff
@@ -0,0 +1,244 @@
...
+例如，代码编辑器和 IDE 本身也可以编辑一大段文本（代码和软件本身也是文本啊），文本编辑器也可以编写代码和软件，
```
Comment (core-man) [inline source/computer/editor.rst:47]:
```suggestion
例如，代码编辑器和 IDE 也可以编辑一大段文本（代码和软件本身也是文本啊），文本编辑器也可以编写代码和软件，
```

File: source/computer/editor.rst [hunk 17]
```diff
@@ -0,0 +1,244 @@
...
+代码编辑器一般也有 IDE 的部分集成开发功能。一般而言，文本编辑器启动速度最快，集成开发环境最慢。
+
+例如，按照以上分类，`vim <https://www.vim.org/>`__ 编辑器是一款基于命令行的文本或代码编辑器，
+`Visual Studio Code <https://code.visualstudio.com/>`__\ （VS Code）是一款图形化的代码编辑器或 IDE，
+`PyCharm <https://www.jetbrains.com/pycharm/>`__ 是一款图形化的IDE（主要开发 Python 语言写的软件）。
+
+需要注意的是，编写代码时，经常在文件间进行切换，并阅读、浏览和修改代码，而不是连续编写一大段的文字，因此编写代码和
```
Comment (core-man) [inline source/computer/editor.rst:54]:
```suggestion
需要注意的是，编写代码时，经常要在文件间进行切换，并阅读、浏览和修改代码，而不是连续编写一大段文字，因此编写代码和
```

File: source/computer/editor.rst [hunk 18]
```diff
@@ -0,0 +1,244 @@
...
+编辑一长段纯文本是很不一样的。编写程序/软件时，还需要进行编译、链接、调试、版本控制等。因此，纯粹的代码编辑器和 IDE 是有所区别的，
+前者主要是让我们优雅地编写代码，后者集成了诸如编译、链接、调试、版本控制等功能，使得开发程序（往往是大型软件）更加高效。
```
Comment (core-man) [inline source/computer/editor.rst:56]:
```suggestion
前者主要是让我们优雅地编写代码，后者集成了诸如编译、链接、调试、版本控制等功能，主要是让我们更高效地开发程序（往往是大型软件）。
```

File: source/computer/editor.rst [hunk 19]
```diff
@@ -0,0 +1,244 @@
...
+
+对于刚接触编辑器的读者，我们根据个人经验推荐一些编辑器。熟练掌握这些编辑器后，可以尝试和使用其他更适合的编辑器。
```
Comment (core-man) [inline source/computer/editor.rst:58]:
```suggestion
对于刚接触编辑器的读者，我们根据个人经验推荐一些编辑器。熟练掌握这些编辑器后，可以尝试和使用其他更适合自己的编辑器。
```

File: source/computer/editor.rst [hunk 20]
```diff
@@ -0,0 +1,244 @@
...
+`Visual Studio Code <https://code.visualstudio.com/>`__\ （VS Code）是一款图形化的代码编辑器或 IDE，
+`PyCharm <https://www.jetbrains.com/pycharm/>`__ 是一款图形化的IDE（主要开发 Python 语言写的软件）。
+
+需要注意的是，编写代码时，经常要在文件间进行切换，并阅读、浏览和修改代码，而不是连续编写一大段文字，因此编写代码和
+编辑一长段纯文本是很不一样的。编写程序/软件时，还需要进行编译、链接、调试、版本控制等。因此，纯粹的代码编辑器和 IDE 是有所区别的，
+前者主要是让我们优雅地编写代码，后者集成了诸如编译、链接、调试、版本控制等功能，主要是让我们更高效地开发程序（往往是大型软件）。
+
+对于刚接触编辑器的读者，我们根据个人经验推荐一些编辑器。熟练掌握这些编辑器后，可以尝试和使用其他更适合的编辑器。
+
+- VS Code 是目前最流行的代码编辑器或 IDE，启动较快，编辑文本也非常高效。\ **推荐**\ 在日常科研中使用 VS Code 编辑文本和代码
```
Comment (core-man) [inline source/computer/editor.rst:60]:
```suggestion
- VS Code 是目前最流行的代码编辑器或 IDE。启动较快，编辑文本也非常高效。\ **推荐**\ 在日常科研中使用 VS Code 编辑文本和代码
```

File: source/computer/editor.rst [hunk 21]
```diff
@@ -0,0 +1,244 @@
...
+
+           $ ssh-copy-id -i ~/.ssh/id_rsa-remote-ssh.pub seismo-learn@192.168.1.100
+
+       该命令会把本地的公钥以追加的方式复制到远程计算机的 :file:`~/.ssh/authorized_keys` 文件中，并给远程计算机
+       中的用户家目录、:file:`~/.ssh` 目录以及 :file:`~/.ssh/authorized_keys` 设置合适的权限。
+       若远程计算机或本地计算机是 Windows 系统，请参考
+       `Quick start: Using SSH keys <https://code.visualstudio.com/docs/remote/troubleshooting#_quick-start-using-ssh-keys>`__
+       进行配置。
+
+       在终端中输入以下命令，验证是否配置是否成功，即可以登录远程计算机（用户需替换自己的远程计算机用户名和 IP 地址）::
```
Comment (core-man) [inline source/computer/editor.rst:125]:
```suggestion
       在终端中输入以下命令，验证是否配置是否成功（即可以登录远程计算机）::
```

File: source/computer/editor.rst [hunk 22]
```diff
@@ -0,0 +1,244 @@
...
+
+在大多数 Linux 发行版中，vim 也可以用 vi 替代。可以使用以下命令查看::
+
+    $ which vim
+    /usr/bin/vim
+
+    $ which vi                          
+    vi=vim
+
+本文只介绍了 vim 的一些常见用法。
```
Comment (core-man) [inline source/computer/editor.rst:183]:
```suggestion
```

File: source/computer/editor.rst [hunk 23]
```diff
@@ -0,0 +1,244 @@
...
+- :kbd:`yy` 复制当前行
+- :kbd:`p` 把复制内容粘贴到当前行的下面
+- :kbd:`dd` 删除当前行
+- :kbd:`u` 撤销上一次修改
+- :kbd:`Ctrl` + :kbd:`r` 重做上一次修改
+
+扩展阅读
+^^^^^^^^^
+
+日常科研中需要经常使用 vim 的用户可以参考以下教程，学习和掌握更多用法、配置 vim 来更高效地编辑文本：
```
Comment (core-man) [inline source/computer/editor.rst:241]:
```suggestion
本文只介绍了 vim 的一些常见用法。日常科研中需要经常使用 vim 的用户可以参考以下教程，
学习和掌握更多用法、配置 vim 来更高效地编辑文本：
```

File: source/computer/editor.rst [hunk 24]
```diff
@@ -0,0 +1,243 @@
...
+- 命令模式：执行一些指令
+
+在不同的操作模式下，键盘敲击的含义不同。Vim 默认会在左下角显示当前的操作模式。启动时的默认模式是正常模式，
+可以使用以下方式，在不同模式之间进行切换：
+
+- 按下 :kbd:`Esc` 键可以从任何其他模式返回正常模式
+- 在正常模式下，按 :kbd:`i` 键进入插入模式
+- 在正常模式下，按 :kbd:`:` 键进入命令模式
+
+日常科研中大部分时间会花在正常模式和插入模式。
```
Comment (core-man) [inline source/computer/editor.rst:199]:
```suggestion
日常科研使用 vim 编辑器时，大部分时间处于正常模式和插入模式中。
```

File: source/computer/editor.rst [hunk 25]
```diff
@@ -0,0 +1,243 @@
...
+
+基本操作
+^^^^^^^^
+
+使用 vim 编辑器打开 :file:`hello-world.txt` 文件，若文件不存在则创建该文件。同时使用以上按键
```
Comment (core-man) [inline source/computer/editor.rst:204]:
```suggestion
使用 vim 编辑器打开 :file:`hello-world.txt` 文件，若文件不存在则创建该文件。使用以上按键
```

File: source/computer/editor.rst [hunk 26]
```diff
@@ -0,0 +1,243 @@
...
+在不同的操作模式之间进行切换，并观察左下角地显示::
```
Comment (core-man) [inline source/computer/editor.rst:205]:
```suggestion
在不同操作模式之间进行切换，并观察左下角的显示::
```

File: source/computer/editor.rst [hunk 27]
```diff
@@ -0,0 +1,243 @@
...
+
+    # 若 vim 后面没有文件名，也会进入 vim
+    $ vim hello-world.txt
+
+按 :kbd:`i` 键进入插入模式，输入 Hello World，然后按 :kbd:`Esc` 键回到正常模式。
+假设这时文本输入完毕，我们可以进入命令模式（按下 :kbd:`:` 键），这时光标会立即跳到屏幕下方的命令行，
+然后输入 :kbd:`wq` 即可保存文件并退出 vim。
```
Comment (core-man) [inline source/computer/editor.rst:212]:
```suggestion
按 :kbd:`i` 键进入插入模式，输入 Hello World，然后按 :kbd:`Esc` 键回到正常模式。
假设文本输入完毕且处于正常模式，按下 :kbd:`:` 键进入命令模式，这时光标会立即跳到屏幕下方的命令行，
然后输入 :kbd:`wq` 即可保存文件并退出 vim。
```

File: source/computer/editor.rst [hunk 28]
```diff
@@ -0,0 +1,243 @@
...
+
+- :kbd:`:q` 退出 vim 并关闭窗口
+- :kbd:`:w` 保存文件
+- :kbd:`:wq` 保存文件并退出 vim
+
+在正常模式下，可以使用以下常用移动命令在文本中快速移动：
+
+- 行： \ :kbd:`0`\ （行初）、\ :kbd:`^`\ （第一个非空格字符）、\ :kbd:`$`\ （行尾）
+- 文件：\ :kbd:`gg`\ （文件头）、\ :kbd:`Shift` + :kbd:`g`\ （文件尾）
+- 行数：{number} + :kbd:`Shift` + :kbd:`g` 会移动至行号为 number（如 20）的行（第 20 行）
```
Comment (core-man) [inline source/computer/editor.rst:224]:
```suggestion
- 行数：number + :kbd:`Shift` + :kbd:`g` 会移动至行号为 number（如 20）的行（即第 20 行）
```

File: source/computer/editor.rst [hunk 29]
```diff
@@ -0,0 +1,243 @@
...
+- :kbd:`p` 把复制内容粘贴到当前行的下面
+- :kbd:`dd` 删除当前行
+- :kbd:`u` 撤销上一次修改
+- :kbd:`Ctrl` + :kbd:`r` 重做上一次修改
+
+扩展阅读
+^^^^^^^^^
+
+本文只介绍了 vim 的一些常见用法。日常科研中需要经常使用 vim 的用户可以参考以下教程，
+学习和掌握更多用法、配置 vim 来更高效地编辑文本：
```
Comment (core-man) [inline source/computer/editor.rst:240]:
```suggestion
学习和掌握更多用法，配置 vim，以便更高效地编辑文本：
```

**Pull request merged**